src/GitReportCL/GitDiffFinalOutputBuilder.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Generic;
+namespace GitReport.CLI
+{
+    class ReportCreator
+    {
+        Dictionary<string, ModificationCounters> dictionaryManager = 
+            new Dictionary<string, ModificationCounters>();
+
+        JsonConfig jsonManager = new JsonConfig();
+        private void CreateReportForComponent(string oneLineFromStdOut)
+        {
+            var componentNewId = string.Empty;
+            string[] separatedGitDiffOutput = oneLineFromStdOut.Split('\t');
+
+            ModificationCounters modArg = new ModificationCounters();
+            modArg.AdditionCounter = CheckIfStringIsNumber(separatedGitDiffOutput[0]);
+            modArg.DeletionCounter = CheckIfStringIsNumber(separatedGitDiffOutput[1]);
+
+            if (jsonManager.TryMatchPaths(separatedGitDiffOutput[2], out componentNewId))
+            {
+                if (CheckIfKeyExists(componentNewId))
+                {
+                    dictionaryManager[componentNewId].AdditionCounter += modArg.AdditionCounter;
+                    dictionaryManager[componentNewId].DeletionCounter += modArg.DeletionCounter;
+                }
+                else
+                {
+                    dictionaryManager.Add(componentNewId, modArg);
+                }
+            }
+        }
+        public Dictionary<string, ModificationCounters> CreateWholeReport(string wholeStdOut)
+        {
+            string[] outputLineByLine = wholeStdOut.Split('\n');
+            for (int i = 0; i < outputLineByLine.Length - 1; i ++)                                                            
+            {
+                CreateReportForComponent(outputLineByLine[i]);
+            }
+            return dictionaryManager;
+        }
+        private bool CheckIfKeyExists(string temporaryKey)
```

Comment:
you could remove whole method and directly call on `dictionaryManager.ContainsKey` inside if statement